propolis/src/block/crucible.rs

```diff
@@ -78,12 +79,14 @@ impl CrucibleBackend {
     }
 
     /// Issue a snapshot request
-    pub fn snapshot(&self, snapshot_name: String) -> Result<()> {
+    pub fn snapshot(&self, snapshot_id: Uuid) -> Result<()> {
         // XXX Crucible uses std::sync::mpsc::Receiver, not
         // tokio::sync::mpsc::Receiver, so use tokio::task::block_in_place here.
         // Remove that when Crucible changes over to the tokio mpsc.
         let mut waiter = tokio::task::block_in_place(|| {
-            self.block_io.flush(Some(SnapshotDetails { snapshot_name }))
+            self.block_io.flush(Some(SnapshotDetails {
+                snapshot_name: snapshot_id.to_string(),
```

Comment:
Nit: I realize this would be on the crucible side but is the snapshot name intended to always be a UUID or any string?

Reply:
`SnapshotDetails` accepts any string, but from Nexus we should accept snapshot UUIDs.